From: Li RongQing <lirongqing@baidu.com>
Date: Fri, 30 Jan 2026 03:56:03 -0500
Subject: [PATCH] mm/damon/stat: remove __read_mostly from
 memory_idle_ms_percentiles

The 'memory_idle_ms_percentiles' array in DAMON_STAT is updated
frequently by the kernel to reflect the latest idle time statistics.
Marking it as '__read_mostly' is inappropriate for data that is
regularly written to, as it can lead to cache pollution in the
read-mostly section.

Remove the '__read_mostly' annotation to accurately reflect the
variable's usage pattern.

Link: https://patch.msgid.link/20260130085603.1814-1-lirongqing@baidu.com
Signed-off-by: Li RongQing <lirongqing@baidu.com>
Reviewed-by: SeongJae Park <sj@kernel.org>
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/stat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/stat.c b/mm/damon/stat.c
index 536f02bd173e3..bcf6c8ae9b906 100644
--- a/mm/damon/stat.c
+++ b/mm/damon/stat.c
@@ -34,7 +34,7 @@ module_param(estimated_memory_bandwidth, ulong, 0400);
 MODULE_PARM_DESC(estimated_memory_bandwidth,
 		"Estimated memory bandwidth usage in bytes per second");
 
-static long memory_idle_ms_percentiles[101] __read_mostly = {0,};
+static long memory_idle_ms_percentiles[101] = {0,};
 module_param_array(memory_idle_ms_percentiles, long, NULL, 0400);
 MODULE_PARM_DESC(memory_idle_ms_percentiles,
 		"Memory idle time percentiles in milliseconds");
-- 
2.47.3

